### DISCUSSION TITLE
Double control

### DISCUSSION THREAD

**Discussion by lpuntaloro** [HomeSpan/HomeSpan#470] [General]
Hello,

Thank you so much again for the beautiful library you have made available.

I have a question... I have lighting in my room with 2 led strips, one with a warm light and one with a cold light. Is there a dedicated accessory on homespan to manage it directly in a single control? If I use dimmable led I have to configure 2 separately...

Thank you

**Comment by HomeSpan** [comment]
Not sure what you mean by a "single control."

If you mean you want to control each light separately, but have both controls be contained within the same Home App Tile, this is easy to do.  You would create two Lightbulb Services in the same Accessory.  See the 4th Accessory listed in Example 11 for details.

On the other hand, if you are looking to combine your two led strips into a single appliance, this is also possible depending on exactly what you are trying to do.  One possibility would be to use the Color Temperature Characteristic of the Lightbulb Service and have HomeSpan fade proportionally between your warm LED and cold LED depending on what temperature you select.  In this case you would user two LedPin classes - one controls the brightness of the wam LED and one controls the brightness of the cold LED.  You'll have to create your own formula for determining how bright to set each one depending on the Color Temperature you choose from the Home App.  If your warm LED was rated at 3000K and your cold LED was rated at 5000K, you could set the range to the Color Temperature variable to be from 3000K to 5000K, and then vary the brightness of each LedPin proportionally.

Note the Color Temperature Characteristic Apple uses is measured in inverse Kelvin, instead of just Kelvin.  See the HAP-R2 documentation for details.

**Comment by lpuntaloro** [comment]
HI,

Thank you for your valuable advice, I have implemented the 2 solutions and now I will choose which one to develop... I attach screenshots!!!!!

Can the color temperature range be set programmatically?

Thank you



Luigi Puntaloro
Il 30 dic 2022, 21:28 +0100, HomeSpan ***@***.***>, ha scritto:

**Comment by HomeSpan** [comment]
Yes, the range of Characteristics with variable values, such as Color Temperature, can be changed within HomeSpan using the `setRange()` method.  See the [API](https://github.com/HomeSpan/HomeSpan/blob/master/docs/Reference.md#spancharacteristicvalue-boolean-nvsstore) for details.

**Comment by lpuntaloro** [comment]
Thank you and happy end of the year!!!!!

Luigi Puntaloro
Il 30 dic 2022, 22:16 +0100, HomeSpan/HomeSpan ***@***.***>, ha scritto:
>
> proportionally

**Comment by HomeSpan** [comment]
Thanks and Happy New Year as well!
